server/controllers/recipesFilterController.js

```diff
@@ -0,0 +1,58 @@
+const Recipe = require('../models/RecipeModel');
+const Ingredient = require('../models/IngredientModel');
+
+// Get all recipes by certain ingredient
+const getRecepiesByIngredients = async (req, res) => {
+  //todo - check what happens in "all" category!!
+
+  try {
+    const { ingredients, category } = req.query;  
+    // Filter only by category - No ingredients were provided
+    if (!ingredients) { 
+      const recipes = await Recipe.find();
+      const filteredRecipes = recipes.filter(recipe => recipe.categories.includes(category));
```

Comment:
You can go directly to the Category database, where you have all the recipes in the category, so you won't have to filter them